tesseract_time_parameterization/src/time_optimal_trajectory_generation.cpp

```diff
@@ -818,10 +827,24 @@ void Trajectory::integrateBackward(std::list<TrajectoryStep>& start_trajectory,
     // Check for intersection between current start trajectory and backward
     // trajectory segments
     const double start_slope = (start2->path_vel_ - start1->path_vel_) / (start2->path_pos_ - start1->path_pos_);
-    const double intersection_path_pos =
-        (start1->path_vel_ - path_vel + slope * path_pos - start_slope * start1->path_pos_) / (slope - start_slope);
-    if (std::max(start1->path_pos_, path_pos) - EPS <= intersection_path_pos &&
-        intersection_path_pos <= EPS + std::min(start2->path_pos_, trajectory.front().path_pos_))
+
+    // It is possible to have both slop and start_slope to be equal
+    bool check_eq_slope = tesseract_common::almostEqualRelativeAndAbs(slope, start_slope, EPS);
+    double intersection_path_pos{ 0 };
+    if (check_eq_slope)
+      intersection_path_pos = start1->path_pos_ + (start2->path_pos_ - start1->path_pos_) / 2.0;
+    else
+      intersection_path_pos =
+          (start1->path_vel_ - path_vel + slope * path_pos - start_slope * start1->path_pos_) / (slope - start_slope);
```

Comment:
this could perhaps do with a comment? What happens if they are equal? And why? What if they are not?

Reply:
> What happens if they are equal? 

This occurs if two consecutive `TrajectorySteps` have the same acceleration. Resulting in `intersection_path_pos` being nan because it divides by zero.

> What if they are not?

I have not looked at the paper to see what is being calculated here.

Reply:
I (perhaps wrongfully) assumed this was added, due to the `+` in the gutter.